### ISSUE TITLE
Trying to refresh a UI Widget inside a Dashboard Widget

### ISSUE THREAD

**Issue by robertmazzo** [DataTorrent/malhar-angular-dashboard#116]
Hello Andy,
I'm trying to update the data bindings to a treegrid UI widget which is placed on a dashboard widget; however, I cannot get this treegrid widget to refresh ! In other words, the initial rendering of my treegrid is fine with my hard coded data; however it does NOT refresh after updating its data bindings from the server.

Here are the steps:
1) Launch the dashboard initially. The treegrid's hard coded data bindings render the grid sucessfully.
2) Launch the modal window from the dashboard widget. This brings up several configuration tabs which the user can choose from. Once saved, the parameters are submitted to the server, the return data is parsed in my controller. Finally I update the treegrid's datasource.
3) Now I attempt to update $scope.result.dataModel.treeGridOptions on the $modalInstance.close event; however I cannot figure out how to refresh the widget content (which should re-render my treegrid widget).

I'm wondering if this problem is more related to the widget ui library I'm using (i.e. I use both Kendo UI and jqWidgets), or if I'm totally missing something within your framework.

Ex/ Here's the my initial grid:

![treegrid-1](https://cloud.githubusercontent.com/assets/8823366/6116294/9b4ed9f4-b079-11e4-83f5-ab842e9a2b8f.png)

Ex/ Here's my modal window where i configure the grid widget, and update the grid's data binding on SAVE event 

![treegrid-modal1](https://cloud.githubusercontent.com/assets/8823366/6116320/d78b689c-b079-11e4-9720-136881ce2be0.png)

My widget def is defined here with the modal options template and controller :

        {
                name: 'chart_treegrid',
                title: 'Report Grid',            
                templateUrl: 'app/shared/gadget-templates/tree-grid.html',
                dataModelType: TreeGridDataModel,
                settingsModalOptions: {   // for grid gadget config popup
                    templateUrl: 'app/shared/include/grid-config/widget-config-grid.html',
                    controller: 'GridSettingsCtrl as settings'
                }
            }

The GridSettingsCtrl controller code looks something like :  

         (function () {
    'use strict';
    angular.module('rage')
       .controller('GridSettingsCtrl', ['$rootScope', '$scope', '$modalInstance', 'datacontext', 'gridHierarchyService', 'reportsContext', 'gadgetHelperFactory', 'widget', gridSettings]);

    function gridSettings($rootScope, $scope, $modalInstance, datacontext, gridHierarchyService, reportsContext, gadgetHelperFactory, widget) {

        var settings = this;

       // add widget to scope
        $scope.widget = widget;

        // set up result object
        $scope.result = jQuery.extend(true, {}, widget);

        $scope.ok = function () {       // SAVE USERS SETTINGS !
            parseGadgetSettingsData();
            //$modalInstance.close($scope.result);  // moved below at end of grid processing.
        };

        $scope.cancel = function () {
            $modalInstance.dismiss('cancel');
        };        

        function parseGadgetSettingsData() {
             // some code omitted for brevity

             var dataAdapter = new $.jqx.dataAdapter(myModel);  // WIRE UP NEW DATA ADAPTER !

            $scope.result.dataModel.treeGridOptions.source = dataAdapter;  // UPDATE RESULT SCOPE
            $scope.result.dataModel.treeGridOptions.columns = colDefs

            $modalInstance.close($scope.result);    // FIRE THE MODAL CLOSE EVENT

            // PRAY MY TREEGRID WIDGET REFRESHES IN MAIN WIDGET CONTAINER ! :-)
        }
      };   // end of gridSettings()    
     })();

In this screen shot you can see my debug window in Chrome. I show $scope.result.dataModel.treeGridOptions , and I can indeed confirm that my data records are refreshed from the server API call. 

![treegrid-result](https://cloud.githubusercontent.com/assets/8823366/6116555/a89c425c-b07b-11e4-983b-3e67f7807bc4.png)

Bottom line: why doesn't my treegrid widget get refreshed ?

**Comment by robertmazzo** [comment]
it appears to be an issue related to the actual UI Gadget's redraw event. 
I've managed to add the `k-rebind` attribute/directive to the Kendo UI div - and it refreshed just fine ! 
I'm now working on the jqWidgets redraw event.
This ticket might be a non-issue for Malhar.

**Comment by andyperlitch** [comment]
I think i follow along, and I'm glad that it is outside the scope of our component. Next time I would do `<pre>{{ dataModel.treeGridOptions | json }}</pre>` to first check that the data is even getting there. Shall we close this one?

**Comment by robertmazzo** [comment]
Yes close

 thx
